File: 833_find_and_replace_in_string/Solution.java
class Solution {
    public String findReplaceString(String s, int[] indices, String[] sources, String[] targets) {
        String newString = s;
        int[] indexAdjustment = new int[s.length()];
        
        for(int i = 0; i<indices.length; i++){
            if(s.indexOf(sources[i], indices[i]) == indices[i]){
                    newString = replaceAtIndex(indices[i] + indexAdjustment[indices[i]], newString, targets[i], sources[i]);
                
                indexAdjustment = updateIndexAdjustments(indices[i],targets[i].length() - sources[i].length(), indexAdjustment);
            }
        }
        
        return newString;
    }
    
    public String replaceAtIndex(int start, String original, String replacement, String toBeReplaced){
        String ret = "";
        ret = original.substring(0, start) + replacement + original.substring(start + toBeReplaced.length());
        
        return ret;
    }
    
    public int[] updateIndexAdjustments(int start, int increment, int[] original){
        for(int i = start; i<original.length; i++){
            original[i] += increment;
        }
        return original;
    }
}

File: 833_find_and_replace_in_string/Solution.cpp
class Solution {
public:
    string findReplaceString(string s, vector<int>& indices, vector<string>& sources, vector<string>& targets) {
        map<int, pair<int, string>, greater<int>> mapping;
        
        for (int i = 0; i < indices.size(); i++){
            if (s.compare(indices[i], sources[i].size(), sources[i]) == 0) {
                mapping[indices[i]] = { sources[i].size(), targets[i] };
            }
        }
        
        for (auto row : mapping){
           s.replace(row.first, row.second.first, row.second.second); 
        } 
        return s;
    }
};
